app/controllers/application.js

```diff
@@ -1,6 +1,24 @@
 import Controller from '@ember/controller';
 import { inject as service } from '@ember/service';
+import { action } from '@ember/object'
 
 export default class ApplicationController extends Controller {
   @service router;
+  
+  @action openExplorerURL() {
+    const explorerURL = "/explorer/cities/NYC"
+    let newURL;
+    if (sessionStorage.length) {
+      let queryParams = new URLSearchParams()
+
+      for (let key in sessionStorage) {
+        queryParams.append(key, sessionStorage[key])
+      }
+
+      newURL = explorerURL + '?' + queryParams.toString()
```

Comment:
This is a relatively small thing but I think it would be better to use the overload of this function that takes the name of a route and an "options" object with the queryParams, instead of "building" your own `URLSearchParams` object here. (See https://api.emberjs.com/ember/3.10/classes/Controller/methods/transitionToRoute?anchor=transitionToRoute)

Reply:
I originally tried using that approach, eg ```this.transitionToRoute('explorer', { queryParams })``` but it wouldn't load the page properly. It would take users to /explorer/**undefined**/**undefined**?compareTo=1 instead of /explorer/cities/NYC?compareTo=1 and I couldn't find a way to pass in those additional URL parameters

Reply:
The "cities" and "NYC" parts of "/explorer/cities/NYC" are [Dynamic Segments](https://guides.emberjs.com/v3.10.0/routing/defining-your-routes/#toc_dynamic-segments) so you have to specify the value for those as separate arguments when you use `transitionToRoute`. (note that dynamic segments aren't quite the same thing as query params but they serve a similar purpose). Try `this.transitionToRoute('explorer', 'cities', 'NYC', { queryParams });`